ReactiveCocoa/Swift/Signal.swift

```diff
@@ -32,17 +34,28 @@ public final class Signal<Value, Error: ErrorProtocol> {
 	///   - generator: A closure that accepts an implicitly created observer
 	///                that will act as an event emitter for the signal.
 	public init(_ generator: @noescape (Observer) -> Disposable?) {
+		state = Atomic(SignalState())
 
 		/// Used to ensure that events are serialized during delivery to observers.
 		let sendLock = Lock()
 		sendLock.name = "org.reactivecocoa.ReactiveCocoa.Signal"
 
-		let generatorDisposable = SerialDisposable()
-
 		/// When set to `true`, the Signal should interrupt as soon as possible.
 		let interrupted = Atomic(false)
 
-		let observer = Observer { event in
+		let observer = Observer { [weak self] event in
+			guard let signal = self else {
+				return
```

Comment:
It can. For example, an implementation of an operator forgot to return the observer disposable as (part of) the generator disposable. Then this input observer would stick in the upstream forever, with its captured `self` being `nil`.